src/Adapter/InMemoryAdapter.php

```diff
@@ -120,6 +121,33 @@ public function loadEventsByMetadataFrom(StreamName $streamName, array $metadata
     }
 
     /**
+     * @param StreamName $streamName
+     * @param DateTimeInterface $since
+     * @param array $metadata
+     * @return ArrayIterator
+     */
+    public function replay(StreamName $streamName, DateTimeInterface $since = null, array $metadata)
+    {
+        if (! isset($this->streams[$streamName->toString()])) {
+            return new ArrayIterator();
+        }
+
+        $streamEvents = [];
+
+        foreach ($this->streams[$streamName->toString()] as $index => $streamEvent) {
+            if (null === $since && $this->matchMetadataWith($streamEvent, $metadata)) {
+                $streamEvents[] = $streamEvent;
+            } elseif((float) $streamEvent->createdAt()->format('U.u') >= (float) $since->format('U.u')
```

Comment:
Damn, or do we need the type cast because PHP does not compare the microseconds correctly?